pages/profile.php

```diff
@@ -163,37 +163,41 @@ function userProfile() {
         }
         $exp_width = round($exp_percent * 2);
 
-        $health_percent = round(($player->health / $player->max_health) * 100);
-        $chakra_percent = round(($player->chakra / $player->max_chakra) * 100);
-        $stamina_percent = round(($player->stamina / $player->max_stamina) * 100);
+        $regen_cut = 0;
+        if($player->battle_id or isset($_SESSION['ai_id'])) {
+            $regen_cut = round(($player->regen_rate + $player->regen_boost) * 0.7, 1);
+        }
+
+        $health_after_regen = ($player->health + ($player->regen_rate + $player->regen_boost - $regen_cut) * 2 > $player->max_health)? $player->max_health : $player->health + (($player->regen_rate + $player->regen_boost - $regen_cut) * 2);
+        $chakra_after_regen = ($player->chakra + ($player->regen_rate + $player->regen_boost - $regen_cut) > $player->max_chakra)? $player->max_chakra : $player->chakra + ($player->regen_rate + $player->regen_boost - $regen_cut);
+        $stamina_after_regen = ($player->stamina + ($player->regen_rate + $player->regen_boost - $regen_cut) > $player->max_stamina)? $player->max_stamina : $player->stamina + ($player->regen_rate + $player->regen_boost - $regen_cut);
+
+        $total_regen_rate = $player->regen_rate + $player->regen_boost - $regen_cut;
 
         echo "<td style='width:50%;'>
-		<label style='width:6.7em;'>Health:</label>" .
-            "<span id='health'>" . sprintf("%.2f", $player->health) . '/' . sprintf("%.2f", $player->max_health) . "</span><br />" .
+		<label style='width:6.7em;' for='healthbar'>Health:</label>" .
+            "<span id='health'>" . sprintf("%.2f", $player->health) .'/' . sprintf("%.2f", $player->max_health) . ($player->health != $player->max_health ? " -> <b style='color: green;'>" . sprintf("%.2f",$health_after_regen) . "</b>" : "") . "</span><br />" .
```

Comment:
Hmm, why do we need to render the health after regen instead of rendering the current health?

Reply:
So it's actually doing both. So you see current health after regen but also the amount you will be at after the next regen. This allows at a glance to see how the regen is applied since there is no indication that the health regen is actually doubled.

Reply:
Hmm maybe the better solution here is to update the regen rate indicator to show the different regen rates for each resource type? Example from WoB: 
![image](https://user-images.githubusercontent.com/1691258/186575962-4ead16ab-84e3-4208-adf4-f1ccce628f7f.png)

Reply:
My concern with showing the "current health -> health after next regen" is it could also be construed as "past health -> current health". By showing regen rate instead it's really clear which is which

Reply:
I can definitely see that concern, personally I like being able to see what my health will be at after I wait some time since I can immediately know if i'll be ready for combat by then or not. I feel like it's pretty clear what your current hp since it's inside the bar while this value points to what it will become.
![image](https://user-images.githubusercontent.com/10229531/186576844-858bb86b-bf1a-4113-aa49-c8156ae858bf.png)

Reply:
Hmm yeah I think you're right, with the color coding and positioning after the max health that is pretty clear 👍 Okay, I'm on board